src/lib/core/evio.c

```diff
@@ -404,19 +404,32 @@ evio_service_bind(struct evio_service *service, const char *uri)
 /** It's safe to stop a service which is not started yet. */
 void
 evio_service_stop(struct evio_service *service)
+{
+	int service_fd = service->ev.fd;
+	evio_service_detach(service);
+	if (service_fd >= 0) {
+		if (close(service_fd) < 0) {
+			say_error("Failed to close socket: %s",
+				  strerror(errno));
+		}
+		if (service->addr.sa_family == AF_UNIX) {
+			if (unlink(((struct sockaddr_un *)
+				   &service->addr)->sun_path) < 0) {
+				say_error("Failed to unlink unix "
+					  "socket path: %s", strerror(errno));
+			}
+		}
+	}
+}
+
+void
+evio_service_detach(struct evio_service *service)
 {
 	say_info("%s: stopped", evio_service_name(service));
```

Comment:
Stop is called only once, but you still print `stopped` multiple times for the same address. Maybe worth moving this into `evio_service_stop()`.

Reply:
Fixed